### PR TITLE
Add code-actions to resource return type

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

The following two code-actions are introduced for resource return type :
1. `Add response content-type` : will add `@http:Payload{mediaType: ""}` annotation to return type

2. `Add response cache configuration` : will add `@http:Cache` annotation to return type

> Fixes [`Create code-actions to add annotations for resource return statement #2662`](https://github.com/ballerina-platform/ballerina-standard-library/issues/2662)

> **Note :** The `Add response content-type` code-action can be further improved by adding the default content-type correspond to the return type

In addition to the code-actions the following compiler errors are also introduced : 
1. Using `@http:Cache` with return type : `error` (or `error?`) will return the following compiler error : 
`invalid usage of cache annotation with return type : 'error'. Cache annotation only supports return types of anydata and SuccessStatusCodeResponse `

> Fixes [`@http:CacheConfig should returns a compiler error when the return type is an error #1825`](https://github.com/ballerina-platform/ballerina-standard-library/issues/1825)

2. Using `@http:Payload` with return type : `error` (or `error?`) will return the following compiler error : 
`invalid usage of payload annotation with return type : 'error'`

## Example

**Code Actions :**

Feb-11-2022 13-02-51

**Compiler Errors :**

```ballerina
import ballerina/http;

service / on new http:Listener(8080) {

    // invalid usage of cache annotation with return type : error. Cache annotation only supports
    // return types of anydata and SuccessStatusCodeResponse
    resource function get helloError() returns @http:Cache error {
        return error("hello");
    }
    
    // invalid usage of payload annotation with return type : error
    resource function get heyError() returns @http:Payload{mediaType: "mediaType"} error {
        return error("hey");
    }
   
    // invalid usage of payload annotation with return type : error?
    // invalid usage of cache annotation with return type : error?. Cache annotation only supports
    // return types of anydata and SuccessStatusCodeResponse
    resource function get greeting(http:Caller caller) returns @http:Payload{mediaType: "mediaType"} 
            @http:Cache error? {
        check caller->respond("Greetings!");
    }
}
```

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: compiler-plugin-tests/src/test/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotationsToReturnTypeTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package io.ballerina.stdlib.http.compiler.codeaction;
```
Comment (chamil321) [inline compiler-plugin-tests/src/test/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotationsToReturnTypeTest.java:1]: Let's add the license header

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddResponseCacheConfigCodeAction.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package io.ballerina.stdlib.http.compiler.codeaction;
```
Comment (chamil321) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddResponseCacheConfigCodeAction.java:1]: Missing license header

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddResponseContentTypeCodeAction.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
+package io.ballerina.stdlib.http.compiler.codeaction;
```
[the remark this answers is not in the document]
Comment (TharmiganK) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddResponseContentTypeCodeAction.java:19]: Fixed in [`ce6e762`](https://github.com/ballerina-platform/module-ballerina-http/pull/887/commits/ce6e7626b19235751370cbba2f5413b124a25352)

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddResponseCacheConfigCodeAction.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            }
+        }
+        if (lineRange == null) {
+            return Collections.emptyList();
+        }
+
+        SyntaxTree syntaxTree = context.currentDocument().syntaxTree();
+        NonTerminalNode node = CodeActionUtil.findNode(syntaxTree, lineRange);
+
+        String mediaTypedPayload = "@http:Cache ";
```
[the remark this answers is not in the document]
Comment (TharmiganK) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddResponseCacheConfigCodeAction.java]: Fixed in [`ce6e762`](https://github.com/ballerina-platform/module-ballerina-http/pull/887/commits/ce6e7626b19235751370cbba2f5413b124a25352)

**Review by chamil321 — APPROVED** [review]

**Pull request merged**